### PR TITLE
Kernel/Graphics: More console fixes

### LINKED ISSUE #7241: Kernel: Consistent "Page fault while handling IRQ!" crash when attempting to switch to framebuffer console.
<details>
<summary>Kernel Crash Log</summary>

```
[#0 colonel(0:0)]: CPU[0] Page fault while handling IRQ! Page fault while handling IRQ! , irq level: 1
[#0 colonel(0:0)]: Kernel regions:
[#0 colonel(0:0)]: BEGIN       END         SIZE        ACCESS  NAME
[#0 colonel(0:0)]: c2001000 -- c2100fff 00100000 RW     kmalloc subheap
[#0 colonel(0:0)]: c2104000 -- c2104fff 00001000 RW     HPET MMIO
[#0 colonel(0:0)]: c2106000 -- c2106fff 00001000 RW     VGA font
[#0 colonel(0:0)]: c2108000 -- c2407fff 00300000 RW     Framebuffer Console
[#0 colonel(0:0)]: c2409000 -- c2a08fff 00600000 RW     Framebuffer
[#0 colonel(0:0)]: c2a0a000 -- c3009fff 00600000 RW     Framebuffer Swap (Blank)
[#0 colonel(0:0)]: c300b000 -- c306afff 00060000 RW     Virtual Console Cells
[#0 colonel(0:0)]: c306c000 -- c30cbfff 00060000 RW     Virtual Console Cells
[#0 colonel(0:0)]: c30cd000 -- c312cfff 00060000 RW     Virtual Console Cells
[#0 colonel(0:0)]: c312e000 -- c318dfff 00060000 RW     Virtual Console Cells
[#0 colonel(0:0)]: c318f000 -- c318ffff 00001000 R    C Signal trampolines
[#0 colonel(0:0)]: c3191000 -- c31a0fff 00010000 RW  T  Kernel stack (thread 0)
[#0 colonel(0:0)]: c31a2000 -- c31b1fff 00010000 RW  T  Kernel stack (thread 1)
[#0 colonel(0:0)]: c31c4000 -- c31d3fff 00010000 RW  T  Kernel stack (thread 3)
[#0 colonel(0:0)]: c31d5000 -- c31e4fff 00010000 RW  T  Kernel stack (thread 4)
[#0 colonel(0:0)]: c31e6000 -- c31e6fff 00001000  W     UHCI Framelist
[#0 colonel(0:0)]: c31e8000 -- c31e9fff 00002000  W     UHCI Queue Head Pool
[#0 colonel(0:0)]: c31eb000 -- c31ecfff 00002000  W     UHCI Transfer Descriptor Pool
[#0 colonel(0:0)]: c31ee000 -- c31fdfff 00010000 RW  T  Kernel stack (thread 5)
[#0 colonel(0:0)]: c31ff000 -- c3202fff 00004000 RW     VirtIO MMIO
[#0 colonel(0:0)]: c3204000 -- c3204fff 00001000 RW     VirtIO Queue
[#0 colonel(0:0)]: c3206000 -- c3206fff 00001000 RW     VirtIO Queue
[#0 colonel(0:0)]: c3208000 -- c3209fff 00002000 RW     VirtIOConsole Receive
[#0 colonel(0:0)]: c320b000 -- c320cfff 00002000 RW     VirtIOConsole Transmit
[#0 colonel(0:0)]: c320e000 -- c3211fff 00004000 RW     VirtIO MMIO
[#0 colonel(0:0)]: c3213000 -- c3213fff 00001000 RW     VirtIO Queue
[#0 colonel(0:0)]: c3215000 -- c3215fff 00001000 RW     VirtIORNG
[#0 colonel(0:0)]: c3217000 -- c3217fff 00001000 RW     E1000 RX
[#0 colonel(0:0)]: c3219000 -- c3219fff 00001000 RW     E1000 TX
[#0 colonel(0:0)]: c321b000 -- c323afff 00020000 RW     E1000 MMIO
[#0 colonel(0:0)]: c323c000 -- c323dfff 00002000 RW     E1000 RX buffer
[#0 colonel(0:0)]: c323f000 -- c3240fff 00002000 RW     E1000 RX buffer
[#0 colonel(0:0)]: c3242000 -- c3243fff 00002000 RW     E1000 RX buffer
[#0 colonel(0:0)]: c3245000 -- c3246fff 00002000 RW     E1000 RX buffer
[#0 colonel(0:0)]: c3248000 -- c3249fff 00002000 RW     E1000 RX buffer
[#0 colonel(0:0)]: c324b000 -- c324cfff 00002000 RW     E1000 RX buffer
[#0 colonel(0:0)]: c324e000 -- c324ffff 00002000 RW     E1000 RX buffer
[#0 colonel(0:0)]: c3251000 -- c3252fff 00002000 RW     E1000 RX buffer
[#0 colonel(0:0)]: c3254000 -- c3255fff 00002000 RW     E1000 RX buffer
[#0 colonel(0:0)]: c3257000 -- c3258fff 00002000 RW     E1000 RX buffer
[#0 colonel(0:0)]: c325a000 -- c325bfff 00002000 RW     E1000 RX buffer
[#0 colonel(0:0)]: c325d000 -- c325efff 00002000 RW     E1000 RX buffer
[#0 colonel(0:0)]: c3260000 -- c3261fff 00002000 RW     E1000 RX buffer
[#0 colonel(0:0)]: c3263000 -- c3264fff 00002000 RW     E1000 RX buffer
[#0 colonel(0:0)]: c3266000 -- c3267fff 00002000 RW     E1000 RX buffer
[#0 colonel(0:0)]: c3269000 -- c326afff 00002000 RW     E1000 RX buffer
[#0 colonel(0:0)]: c326c000 -- c326dfff 00002000 RW     E1000 RX buffer
[#0 colonel(0:0)]: c326f000 -- c3270fff 00002000 RW     E1000 RX buffer
[#0 colonel(0:0)]: c3272000 -- c3273fff 00002000 RW     E1000 RX buffer
[#0 colonel(0:0)]: c3275000 -- c3276fff 00002000 RW     E1000 RX buffer
[#0 colonel(0:0)]: c3278000 -- c3279fff 00002000 RW     E1000 RX buffer
[#0 colonel(0:0)]: c327b000 -- c327cfff 00002000 RW     E1000 RX buffer
[#0 colonel(0:0)]: c327e000 -- c327ffff 00002000 RW     E1000 RX buffer
[#0 colonel(0:0)]: c3281000 -- c3282fff 00002000 RW     E1000 RX buffer
[#0 colonel(0:0)]: c3284000 -- c3285fff 00002000 RW     E1000 RX buffer
[#0 colonel(0:0)]: c3287000 -- c3288fff 00002000 RW     E1000 RX buffer
[#0 colonel(0:0)]: c328a000 -- c328bfff 00002000 RW     E1000 RX buffer
[#0 colonel(0:0)]: c328d000 -- c328efff 00002000 RW     E1000 RX buffer
[#0 colonel(0:0)]: c3290000 -- c3291fff 00002000 RW     E1000 RX buffer
[#0 colonel(0:0)]: c3293000 -- c3294fff 00002000 RW     E1000 RX buffer
[#0 colonel(0:0)]: c3296000 -- c3297fff 00002000 RW     E1000 RX buffer
[#0 colonel(0:0)]: c3299000 -- c329afff 00002000 RW     E1000 RX buffer
[#0 colonel(0:0)]: c329c000 -- c329dfff 00002000 RW     E1000 TX buffer
[#0 colonel(0:0)]: c329f000 -- c32a0fff 00002000 RW     E1000 TX buffer
[#0 colonel(0:0)]: c32a2000 -- c32a3fff 00002000 RW     E1000 TX buffer
[#0 colonel(0:0)]: c32a5000 -- c32a6fff 00002000 RW     E1000 TX buffer
[#0 colonel(0:0)]: c32a8000 -- c32a9fff 00002000 RW     E1000 TX buffer
[#0 colonel(0:0)]: c32ab000 -- c32acfff 00002000 RW     E1000 TX buffer
[#0 colonel(0:0)]: c32ae000 -- c32affff 00002000 RW     E1000 TX buffer
[#0 colonel(0:0)]: c32b1000 -- c32b2fff 00002000 RW     E1000 TX buffer
[#0 colonel(0:0)]: c32b4000 -- c32b4fff 00001000 RW     IDE PRDT
[#0 colonel(0:0)]: c32b6000 -- c32b6fff 00001000 RW     IDE DMA region
[#0 colonel(0:0)]: c32b8000 -- c32b8fff 00001000 RW     IDE PRDT
[#0 colonel(0:0)]: c32ba000 -- c32bafff 00001000 RW     IDE DMA region
[#0 colonel(0:0)]: c32bc000 -- c32bdfff 00002000 RW     AHCI HBA
[#0 colonel(0:0)]: c32bf000 -- c32bffff 00001000 RW     AHCI Port Command List
[#0 colonel(0:0)]: c32c1000 -- c32c1fff 00001000 RW     AHCI Port Command List
[#0 colonel(0:0)]: c32c3000 -- c32c3fff 00001000 RW     AHCI Port Command List
[#0 colonel(0:0)]: c32c5000 -- c32c5fff 00001000 RW     AHCI Port Command List
[#0 colonel(0:0)]: c32c7000 -- c32c7fff 00001000 RW     AHCI Port Command List
[#0 colonel(0:0)]: c32c9000 -- c32c9fff 00001000 RW     AHCI Port Command List
[#0 colonel(0:0)]: c32cb000 -- c32cbfff 00001000 RW     Ext2FS: Block group descriptors
[#0 colonel(0:0)]: c32cd000 -- c59dcfff 02710000 RW     KBuffer
[#0 colonel(0:0)]: c59de000 -- c5a2cfff 0004f000 RW     KBuffer
[#0 colonel(0:0)]: c5a2e000 -- c5a2efff 00001000 RW     Ext2FS: Cached bitmap block
[#0 colonel(0:0)]: c5a30000 -- c5a30fff 00001000 RW     Ext2FS: Cached bitmap block
[#0 colonel(0:0)]: c5a32000 -- c5a41fff 00010000 RW  T  Kernel stack (thread 6)
[#0 colonel(0:0)]: c5a43000 -- c5a52fff 00010000 RW  T  Kernel stack (thread 7)
[#0 colonel(0:0)]: c5a54000 -- c5a63fff 00010000 RW     Kernel Packet Buffer
[#0 colonel(0:0)]: c5a65000 -- c5a84fff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c5a86000 -- c5aa5fff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c5aa7000 -- c5ac6fff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c5ac8000 -- c5ae7fff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c5ae9000 -- c5b08fff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c5b0a000 -- c5b29fff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c5b2b000 -- c5b4afff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c5b4c000 -- c5b6bfff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c5b6d000 -- c5b8cfff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c5b8e000 -- c5badfff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c5baf000 -- c5bcefff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c5bd0000 -- c5beffff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c5bf1000 -- c5c10fff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c5c12000 -- c5c31fff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c5c33000 -- c5c52fff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c5c54000 -- c5c73fff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c5c75000 -- c5c94fff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c5c96000 -- c5cb5fff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c5cb7000 -- c5cd6fff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c5cd8000 -- c5cf7fff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c5cf9000 -- c5d18fff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c5d1a000 -- c5d39fff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c5d3b000 -- c5d5afff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c5d5c000 -- c5d7bfff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c5d7d000 -- c5d9cfff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c5d9e000 -- c5dbdfff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c5dbf000 -- c5ddefff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c5de0000 -- c5dfffff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c5e01000 -- c5e20fff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c5e22000 -- c5e41fff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c5e43000 -- c5e62fff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c5e64000 -- c5e83fff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c5e85000 -- c5ea4fff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c5ea6000 -- c5eb5fff 00010000 RW  T  Kernel stack (thread 9)
[#0 colonel(0:0)]: c5eb7000 -- c5ec6fff 00010000 RW  T  Kernel stack (thread 10)
[#0 colonel(0:0)]: c5ec8000 -- c5ed7fff 00010000 RW  T  Kernel stack (thread 11)
[#0 colonel(0:0)]: c5ed9000 -- c5ee8fff 00010000 RW  T  Kernel stack (thread 12)
[#0 colonel(0:0)]: c5efb000 -- c5f0afff 00010000 RW  T  Kernel stack (thread 14)
[#0 colonel(0:0)]: c5f0c000 -- c5f1bfff 00010000 RW  T  Kernel stack (thread 15)
[#0 colonel(0:0)]: c5f1d000 -- c5f2cfff 00010000 RW  T  Kernel stack (thread 16)
[#0 colonel(0:0)]: c5f2e000 -- c5f3dfff 00010000 RW  T  Kernel stack (thread 17)
[#0 colonel(0:0)]: c5f3f000 -- c5f4efff 00010000 RW  T  Kernel stack (thread 18)
[#0 colonel(0:0)]: c5f50000 -- c5f5ffff 00010000 RW  T  Kernel stack (thread 19)
[#0 colonel(0:0)]: c5f61000 -- c5f70fff 00010000 RW  T  Kernel stack (thread 20)
[#0 colonel(0:0)]: c5f72000 -- c5f81fff 00010000 RW  T  Kernel stack (thread 21)
[#0 colonel(0:0)]: c5f83000 -- c5f92fff 00010000 RW  T  Kernel stack (thread 22)
[#0 colonel(0:0)]: c5f94000 -- c5fa3fff 00010000 RW  T  Kernel stack (thread 23)
[#0 colonel(0:0)]: c5fa5000 -- c5fc4fff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c5fc6000 -- c5fe5fff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c5fe7000 -- c6006fff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c6008000 -- c6027fff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c6029000 -- c6048fff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c604a000 -- c6069fff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c606b000 -- c608afff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c608c000 -- c60abfff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c60ad000 -- c612cfff 00080000 RW     DoubleBuffer
[#0 colonel(0:0)]: c612e000 -- c61adfff 00080000 RW     DoubleBuffer
[#0 colonel(0:0)]: c61af000 -- c61cefff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c61d0000 -- c61effff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c61f1000 -- c6210fff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c6212000 -- c6231fff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c6233000 -- c6252fff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c6254000 -- c6273fff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c6275000 -- c6294fff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c6296000 -- c62b5fff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c62b7000 -- c62d6fff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c62d8000 -- c62f7fff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c62f9000 -- c6318fff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c631a000 -- c6339fff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c633b000 -- c635afff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c635c000 -- c637bfff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c637d000 -- c639cfff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c639e000 -- c63bdfff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c63bf000 -- c63defff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c63e0000 -- c63fffff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c6401000 -- c6420fff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c6422000 -- c6441fff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c6443000 -- c6462fff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c6464000 -- c6483fff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c6485000 -- c64a4fff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c64a6000 -- c64c5fff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c64c7000 -- c64e6fff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c64e8000 -- c6507fff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c6509000 -- c6528fff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c652a000 -- c6549fff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c654b000 -- c656afff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c656c000 -- c658bfff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c658d000 -- c65acfff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c65ae000 -- c65cdfff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c65cf000 -- c65eefff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c65f0000 -- c660ffff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c6611000 -- c6630fff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c6632000 -- c6651fff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c6653000 -- c6672fff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c6685000 -- c66a4fff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c66a6000 -- c66c5fff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c66c7000 -- c66d6fff 00010000 RW  T  Kernel stack (thread 25)
[#0 colonel(0:0)]: c66d8000 -- c6757fff 00080000 RW     DoubleBuffer
[#0 colonel(0:0)]: c31b3000 -- c31b3fff 00001000 RW     Ext2FS: Cached bitmap block
[#0 colonel(0:0)]: c6759000 -- c6778fff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c677a000 -- c6799fff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c679b000 -- c67bafff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c67bc000 -- c67dbfff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c67dd000 -- c67fcfff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c67fe000 -- c681dfff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c681f000 -- c683efff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c6840000 -- c685ffff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c6861000 -- c6880fff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c6882000 -- c68a1fff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c68a3000 -- c68c2fff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c68c4000 -- c68e3fff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c68e5000 -- c6904fff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c6906000 -- c6925fff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c6927000 -- c6946fff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c6948000 -- c6967fff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c6969000 -- c6988fff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c698a000 -- c6999fff 00010000 RW  T  Kernel stack (thread 26)
[#0 colonel(0:0)]: c69ac000 -- c69cbfff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c69cd000 -- c69ecfff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c31b5000 -- c31b5fff 00001000 RW     KBuffer
[#0 colonel(0:0)]: c69ee000 -- c6a6dfff 00080000 RW     DoubleBuffer
[#0 colonel(0:0)]: c6a6f000 -- c6aeefff 00080000 RW     DoubleBuffer
[#0 colonel(0:0)]: c31b7000 -- c31b7fff 00001000 RW     Ext2FS: Cached bitmap block
[#0 colonel(0:0)]: c6af0000 -- c6b0ffff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c6b11000 -- c6b30fff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c6b32000 -- c6b51fff 00020000 RW     DoubleBuffer
[#0 colonel(0:0)]: c6b53000 -- c6c52fff 00100000 RW     kmalloc subheap
[#0 colonel(0:0)]: c6c54000 -- c7053fff 00400000 RW     KBuffer
[#0 colonel(0:0)]: c7055000 -- c7454fff 00400000 RW     KBuffer
[#0 colonel(0:0)]: c31b9000 -- c31b9fff 00001000 RW
[#0 colonel(0:0)]: Unrecoverable page fault, read from address V0xc2a09000
[colonel(0:0)]: CRASH: CPU #0 Page Fault in ring 0
[#0 colonel(0:0)]: Exception code: 0000 (isr: 0000)
[#0 colonel(0:0)]:     pc=0008:c018e029 eflags=00010086
[#0 colonel(0:0)]:  stack=0010:c31a0c34
[#0 colonel(0:0)]:     ds=0010 es=0010 fs=0030 gs=0000
[#0 colonel(0:0)]:    eax=c2a0a000 ebx=c0122498 ecx=004fd000 edx=00afc800
[#0 colonel(0:0)]:    ebp=c31a0cac esp=c31a0c34 esi=c2a09000 edi=c300a000
[#0 colonel(0:0)]:    cr0=80010013 cr2=c2a09000 cr3=0187e000 cr4=00340ee4
[colonel(0:0)]: KERNEL PANIC! :^(
[colonel(0:0)]: Crash in ring 0
[colonel(0:0)]: at ../../Kernel/Arch/i386/CPU.cpp:171 in void Kernel::handle_crash(Kernel::RegisterState&, const char*, int, bool)
[#0 colonel(0:0)]: 0xc03a3766  Kernel::__panic(char const*, unsigned int, char const*) +0xb0
[#0 colonel(0:0)]: 0xc05b7dfd  Kernel::handle_crash(Kernel::RegisterState&, char const*, int, bool) +0x1de
[#0 colonel(0:0)]: 0xc05b8bad  page_fault_handler +0xa6c
[#0 colonel(0:0)]: 0xc05af114  page_fault_asm_entry +0x26
[#0 colonel(0:0)]: 0xc018c3b7  Kernel::BochsGraphicsAdapter::enable_consoles() +0x43d
[#0 colonel(0:0)]: 0xc0190f81  Kernel::GraphicsManagement::deactivate_graphical_mode() +0x3f1
[#0 colonel(0:0)]: 0xc04d5b29  Kernel::ConsoleManagement::switch_to(unsigned int) +0xd4d
[#0 colonel(0:0)]: 0xc017ff4f  Kernel::PS2KeyboardDevice::irq_handle_byte_read(unsigned char) +0xaa7
[#0 colonel(0:0)]: 0xc017940b  Kernel::I8042Controller::irq_process_input_buffer(Kernel::HIDDevice::Type) +0x5e9
[#0 colonel(0:0)]: 0xc017f2c5  Kernel::PS2KeyboardDevice::handle_irq(Kernel::RegisterState const&) +0x137
[#0 colonel(0:0)]: 0xc05bcd11  handle_interrupt +0x6fd
[#0 colonel(0:0)]: 0xc05be415  interrupt_common_asm_entry +0x2b
[#0 colonel(0:0)]: 0xc05af249  exit_kernel_thread +0x0
```
</details>

Steps to reproduce the issue:
- Sync master
- Build then Boot the system
- Type: Alt + 2
- The system will crash.

Others have reported this working, however I have a consistent repro when running on QEMU (kvm) 6.0 and 5.0.

### PR DESCRIPTION
**Pull request by supercomputer7**
Instead of processing the input after receiving an IRQ, we shift the
responsibility to the io work queue to handle this for us, so if a page fault
occurs, the kernel can handle that.

Might fix #7241 (?)

cc @gunnarbeutner @bgianfo

### CODE REVIEW

File: Kernel/Devices/HID/PS2KeyboardDevice.cpp [hunk 1]
```diff
@@ -86,7 +87,9 @@ void PS2KeyboardDevice::handle_irq(const RegisterState&)
 {
     // The controller will read the data and call irq_handle_byte_read
     // for the appropriate device
-    m_i8042_controller->irq_process_input_buffer(HIDDevice::Type::Keyboard);
+    g_io_work->queue([this]() {
+        m_i8042_controller->irq_process_input_buffer(HIDDevice::Type::Keyboard);
```
Comment (tomuta) [inline Kernel/Devices/HID/PS2KeyboardDevice.cpp:91]: I don't think that's going to work right. I think the 8i042's `irq_process_input_buffer` must be called from inside the IRQ handler...

Comment (tomuta) [reply Kernel/Devices/HID/PS2KeyboardDevice.cpp:91]: I think only the call to `ConsoleManagement::switch_to` needs to be called from the worker queue.

Comment (supercomputer7) [reply Kernel/Devices/HID/PS2KeyboardDevice.cpp:91]: Why? what is limiting us from handling input outside of the IRQ?

Comment (tomuta) [reply Kernel/Devices/HID/PS2KeyboardDevice.cpp:91]: > Why? what is limiting us from handling input outside of the IRQ?

EOI?

Comment (tomuta) [reply Kernel/Devices/HID/PS2KeyboardDevice.cpp:91]: The code in the `irq_` was meant to run inside the IRQ handler. But I'm no i8042 expert (I struggled with it too much), so maybe that's ok. But I thought the packets had to be read inside the IRQ handler.

Comment (tomuta) [reply Kernel/Devices/HID/PS2KeyboardDevice.cpp:91]: In fact it has a `VERIFY(Processor::current().in_irq());` right in the first line ;) If that isn't triggering then we have yet another bug ;)

Comment (supercomputer7) [reply Kernel/Devices/HID/PS2KeyboardDevice.cpp:91]: I changed that line, so it worked (it asserted) until I changed it. Maybe we should only let the switch to happen in an IO work queue then.

Comment (tomuta) [reply Kernel/Devices/HID/PS2KeyboardDevice.cpp:91]: Yeah I think only the call to `ConsoleManagement::switch_to`

**Comment by gunnarbeutner** [comment]
Switching to the text console works now. The text output is somewhat misaligned but there's no crash anymore.

Changing the resolution via Display Settings doesn't work yet. Upon changing the resolution WindowServer crashes - in my case both when switching to a larger as well as a smaller resolution:

```
[#0 WindowServer(23:23)]: BochsGraphicsAdapter: resolution set to 1280x1024
[WindowServer(23:23)]: ASSERTION FAILED: !m_userspace_framebuffer_region
[WindowServer(23:23)]: ../../Kernel/Graphics/FramebufferDevice.cpp:35 in virtual Kernel::KResultOr<Kernel::Region*> Kernel::FramebufferDevice::mmap(Kernel::Process&, Kernel::FileDescription&, const Kernel::Range&, u64, int, bool)
[#0 WindowServer(23:23)]: 0xc05c116e  abort +0x4b
[#0 WindowServer(23:23)]: 0xc05b447f  debug_handler +0x0
[#0 WindowServer(23:23)]: 0xc018f844  Kernel::FramebufferDevice::mmap(Kernel::Process&, Kernel::FileDescription&, Kernel::Range const&, unsigned long long, int, bool) +0x708
[#0 WindowServer(23:23)]: 0xc024e749  Kernel::FileDescription::mmap(Kernel::Process&, Kernel::Range const&, unsigned long long, int, bool) +0x271
[#0 WindowServer(23:23)]: 0xc04551b2  Kernel::Process::sys$mmap(AK::Userspace<Kernel::Syscall::SC_mmap_params const*>) +0x2bb8
[#0 WindowServer(23:23)]: 0xc03e08e0  syscall_handler +0x1e7d
[#0 WindowServer(23:23)]: 0xc03dea2b  syscall_asm_entry +0x31
```

**Comment by gunnarbeutner** [comment]
Switching consoles/screens works as does changing the resolution. Very nice. Here's a fun testcase though:
1. Open Display Settings and change the resolution to something else. Don't confirm the change.
2. Switch to a text console.
3. Wait until the desktop reappears.
4. Switch to the desktop with Ctrl-1 again (apparently the system thinks you're still in the text console) and notice how the output is garbled.

**Comment by gunnarbeutner** [comment]
Retested and after switching back with Ctrl-1 the output is no longer garbled. :partying_face:
When Display Settings changes the resolution back after the timeout I'm getting a black screen though.

**Comment by supercomputer7** [comment]
cc @awesomekling 
This seems like it works fine now. I'd be happy to receive feedback and suggestions for fixing problems that are appearing in the changes :)

File: Kernel/TTY/VirtualConsole.cpp [hunk 2]
```diff
@@ -134,12 +134,54 @@ UNMAP_AFTER_INIT void VirtualConsole::initialize()
...
+
+    if (rows() < old_rows_count) {
+        m_lines.shrink(rows());
+    } else {
+        for (size_t row = 0; row < (size_t)(rows() - old_rows_count); row++) {
+            m_lines.append({ true, 0 });
+        }
+    }
+
+    // FIXME: Potential loss of displayed data occur when resolution width shrinks.
```
Comment (awesomekling) [inline Kernel/TTY/VirtualConsole.cpp:163]: That's totally fine. If someone is changing the resolution, there's a 100% chance they will redraw the screen afterwards anyway :)

File: Kernel/Graphics/FramebufferDevice.cpp [hunk 3]
```diff
@@ -40,9 +41,15 @@ KResultOr<Region*> FramebufferDevice::mmap(Process& process, FileDescription&, c
     m_real_framebuffer_region = MM.allocate_kernel_region_with_vmobject(*m_real_framebuffer_vmobject, page_round_up(framebuffer_size_in_bytes()), "Framebuffer", Region::Access::Read | Region::Access::Write);
     m_swapped_framebuffer_region = MM.allocate_kernel_region_with_vmobject(*m_swapped_framebuffer_vmobject, page_round_up(framebuffer_size_in_bytes()), "Framebuffer Swap (Blank)", Region::Access::Read | Region::Access::Write);
 
+    RefPtr<VMObject> choesen_vmobject;
```
Comment (awesomekling) [inline Kernel/Graphics/FramebufferDevice.cpp:44]: Typo, choesen -> chosen

**Comment by gunnarbeutner** [comment]
I'm no longer getting a black screen with the latest changes. :)

File: Kernel/Graphics/FramebufferDevice.h [hunk 4]
```diff
@@ -59,6 +59,8 @@ class FramebufferDevice : public BlockDevice {
     OwnPtr<Region> m_real_framebuffer_region;
     OwnPtr<Region> m_swapped_framebuffer_region;
 
+    bool m_enabled { true };
```
Comment (awesomekling) [inline Kernel/Graphics/FramebufferDevice.h:62]: This name seems confusingly generic. Can we call it something like `m_graphics_mode_enabled` or if you can think of a good descriptive name for what it means to be "enabled" here :)

**Pull request merged**